shared/src/commonMain/kotlin/io/customer/shared/sdk/config/BackgroundQueueConfig.kt

```diff
@@ -0,0 +1,61 @@
+package io.customer.shared.sdk.config
+
+/**
+ * Configurations to hold background queue settings.
+ */
+class BackgroundQueueConfig private constructor(
+    val batchDelayMinTasks: Int,
+    val batchDelayMaxDelayInSeconds: Int,
+    val batchTasksMax: Int,
+) {
+    class Builder {
```

Comment:
Multiple constructors cannot replicate builder pattern. e.g. users cannot provide only `batchDelayMaxDelayInSeconds` or `batchTasksMax` using multiple constructors because all parameters are integers here.